src/main/groovy/com/twcable/gradle/cqpackage/CqPackageUtils.groovy

```diff
@@ -0,0 +1,216 @@
+
+import static java.util.Collections.EMPTY_LIST
+import static java.util.Collections.EMPTY_SET
+
+@Slf4j
+@CompileStatic
+class CqPackageUtils {
+    private static final Pattern JAR_NAME_PATTERN = Pattern.compile(/(?i).*\.jar$/)
+
+    /**
+     * Returns all JAR files generated by the given project and its project dependencies in
+     * the configuration that use the "osgi" plugin.
+     */
+    @Nonnull
+    static Collection<File> projectBundleJarFile(@Nonnull Project project, @Nonnull Configuration configuration) {
+        if (configuration == null) throw new IllegalArgumentException("configuration == null")
+
+        Collection<ProjectDependency> allOsgiProjects = projectDependenciesForConfiguration(configuration).
+            findAll { isOsgiProject(it.dependencyProject) }
+
+        log.debug "osgi projects: ${allOsgiProjects}"
+
+        Collection<File> allProjectJars = jarsForProjectDependencies(allOsgiProjects)
+
+        // if the project passed also generates a bundle, add its jar(s) to the result
+        if (isOsgiProject(project)) {
+            allProjectJars.addAll(jarsForArtifacts(configuration))
+        }
+
+        log.debug "OSGi project bundles: ${allProjectJars.sort()}"
+
+        return allProjectJars
+    }
+
+    /**
+     * Does the project use the "osgi" plugin?
+     */
+    static boolean isOsgiProject(Project project) {
+        if (project == null) throw new IllegalArgumentException("project == null")
+        return project.plugins.findPlugin('osgi') != null
+    }
+
+    /**
+     * Returns all jar files generated by the project dependencies in the configuration that use the "osgi" plugin.
+     */
+    @Nonnull
+    static Collection<File> projectJars(@Nonnull Configuration configuration) {
+        if (configuration == null) throw new IllegalArgumentException("configuration == null")
+
+        def projectDependencies = projectDependenciesForConfiguration(configuration)
+        def allProjectJars = jarsForProjectDependencies(projectDependencies)
+
+        log.debug "Project JARs: ${allProjectJars.sort()}"
+
+        return allProjectJars
+    }
+
+
+    @Nonnull
+    private static Collection<File> jarsForProjectDependencies(Collection<ProjectDependency> projectDependencies) {
+        return projectDependencies.collectMany { ProjectDependency dep ->
+            jarsForArtifacts(dep.projectConfiguration)
+        }
+    }
+
+
+    @Nonnull
+    static Collection<File> jarsForArtifacts(@Nonnull Configuration configuration) {
+        if (configuration == null) throw new IllegalArgumentException("configuration == null")
+        def allArtifacts = configuration.allArtifacts
+        def fileCollection = allArtifacts.files
+        def setOfFiles = fileCollection.files
+        setOfFiles.findAll { JAR_NAME_PATTERN.matcher(it.name).matches() }
+    }
+
+
+    @Nonnull
+    private static Collection<ProjectDependency> projectDependenciesForConfiguration(Configuration configuration) {
```

Comment:
It just made the declaration line too long (>120 chars) and since it's a private method it doesn't really add much.